### ISSUE TITLE
podman cp via STDIN leaves temporary files at tmpdir

### ISSUE THREAD

**Issue by alexshkatov** [containers/podman#19496]
### Issue Description

When a file is copied to a running container with `podman cp` by streaming a tar archive from STDIN, a temporary file is created https://github.com/containers/podman/blob/bde942e21622d489125e3e190988b342e4e7d852/cmd/podman/containers/cp.go#L381
This file is not deleted when the operation is completed causing tmpdir to fills up eventually.

### Steps to reproduce the issue

Steps to reproduce the issue
1. Start a test container
```
$ podman run -dt -p 8080:80/tcp docker.io/library/httpd
67c085b8f99fbbe72c86c8d6077d586627c8521858e247a2017a5aaf3cec9d09
```
2. Create a tar.gz file and copy it into the test container
```
$ ls -l ls.tar.gz
-rw-rw-r--. 1 centos centos 66545 Aug  3 12:09 ls.tar.gz
$ podman cp - 67c085b8f99f:/ < ls.tar.gz
```
3. When the operation is finished, temporary file is still present at tempdir
```
$ ls -l /var/tmp
total 340
-rw-------. 1 centos centos 66545 Aug  3 12:11 2002620236
-rw-------. 1 centos centos 66545 Aug  3 12:11 2711454316
-rw-------. 1 centos centos 66545 Aug  3 12:11 3695615676
-rw-------. 1 centos centos 66545 Aug  3 12:11 552526117
-rw-------. 1 centos centos 66545 Aug  3 12:11 623461537
```

### Describe the results you received

Garbage files are present at tempdir
```
$ ls -l /var/tmp
total 340
-rw-------. 1 centos centos 66545 Aug  3 12:11 2002620236
-rw-------. 1 centos centos 66545 Aug  3 12:11 2711454316
-rw-------. 1 centos centos 66545 Aug  3 12:11 3695615676
-rw-------. 1 centos centos 66545 Aug  3 12:11 552526117
-rw-------. 1 centos centos 66545 Aug  3 12:11 623461537
```

### Describe the results you expected

I'd expect podman to delete the temporary files when copy operation is finished.

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.30.0
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.7-1.module_el8+454+d7ef4b8d.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: 6aaf993c7765d6579b6c7d9178f2b00b1691c1f9'
  cpuUtilization:
    idlePercent: 93.01
    systemPercent: 0.94
    userPercent: 6.05
  cpus: 1
  databaseBackend: boltdb
  distribution:
    distribution: '"centos"'
    version: "8"
  eventLogger: file
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 4.18.0-408.el8.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 109010944
  memTotal: 840695808
  networkBackend: cni
  ociRuntime:
    name: runc
    package: runc-1.1.7-2.module_el8+463+b24391b1.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.1.7
      spec: 1.0.2-dev
      go: go1.20.4
      libseccomp: 2.5.2
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_SYS_CHROOT,CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-3.module_el8+454+d7ef4b8d.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 0
  swapTotal: 0
  uptime: 0h 30m 31.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/centos/.config/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 1
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/centos/.local/share/containers/storage
  graphRootAllocated: 42938118144
  graphRootUsed: 2159763456
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/centos/.local/share/containers/storage/volumes
version:
  APIVersion: 4.5.1
  Built: 1686823293
  BuiltTime: Thu Jun 15 10:01:33 2023
  GitCommit: ""
  GoVersion: go1.20.4
  Os: linux
  OsArch: linux/amd64
  Version: 4.5.1
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by Luap99** [comment]
Thanks for the detailed report, are you interested in opening a PR to fix it?

**Comment by swaragade** [comment]
Hi,
Is there any ETA for this fix to be released ?

Thanks.

**Comment by rhatdan** [comment]
Podman 4.7 is scheduled to be released in October.

**Comment by swaragade** [comment]
Hi,

Can you please confirm if this fix was released in 4.7.X ?

I see: 
`Temporary files created when dealing with images in /var/tmp will automatically be cleaned up on reboot.`
Does it mean, that users need to restart the podman for cleanup ?

Thank you.

**Comment by rhatdan** [comment]
They should be cleaned up right away not at reboot.
